### ISSUE TITLE
Reed is likely aware but I am able to make the map spazz out and display a blank screen by jumping around between views in the app. Main Map to Other back to Main Map back to Sync back to Main Map then Planned Pending Filter...boom

### ISSUE THREAD

**Issue by HATSBot** [WSDOT/HATS#1022]
* **Device Name:** HQD6908002
* **Device Model:** iPad Pro (9.7-inch)
* **Device System Version:** 11.3.1
* **Application Version:** HATS-iOS.beta 4.0.0B1
* **Name:** Joe Schmit
* **Has Attachments:** True

* **Feedback:** Reed is likely aware but I am able to make the map spazz out and display a blank screen by jumping around between views in the app. Main Map to Other back to Main Map back to Sync back to Main Map then Planned Pending Filter...boom

* **Details With Attachments:**  <a href='http://hatsqa.wsdot.loc/Maintenance/Management/Activities/Feedback/GetMobileDeviceFeedback?mobileFeedbackId=1826'>View Details And Attachments</a>

**Comment by ReedHunter-Wsdot** [comment]
With the planned/pending filter fixed, the map no long goes boom, but the map is still blank.  Looking into this.

**Comment by ReedHunter-Wsdot** [comment]
Clicking around is only blanking the map for me if I have a pending activity waiting to sync. 
**Steps to recreate ...**
1. switch the map to planned/pending.
2. inspect a feature so the hasPendingActivity flag is cleared.  The feature will disappear from the planned/pending view.
3. Tap the Sync tab.
4. Tap [Open Sync]
5. Tap [Done].  Don't even bother syncing.
**Expected result:** 
     Map should reload with all the data visible.
**Result**
     Map only has grey squares loading, no data.

**Comment by ReedHunter-Wsdot** [comment]
The planned/pending filter was broken since the new Sdk handles that a little differently.  I fixed that about 2 weeks ago, taking care of the "Boom" part of this bug.  I also had to fix the startup sequence for managing map layers since the new Sdk works so differently.  We used to just recreate the map from the datasource every time we'd go to the map tab.  Now the Sdk doesn't let you do that, so I cache and reuse the layers if we don't need to recreate them.  It's a better design this way in any case, and took care of the blank screen part of this bug.
